mypy/test/data/check-classes.test

```diff
@@ -1418,3 +1418,215 @@ C(0)
 C(arg=0)
 [out]
 
+
+-- Type[C]
+-- -------
+
+
+[case testTypeUsingTypeCBasic]
+from typing import Type
+class User: pass
+class ProUser(User): pass
+def new_user(user_class: Type[User]) -> User:
+    return user_class()
+reveal_type(new_user(User))  # E: Revealed type is '__main__.User'
+reveal_type(new_user(ProUser))  # E: Revealed type is '__main__.User'
+[out]
+
+[case testTypeUsingTypeCTypeVar]
+from typing import Type, TypeVar
+class User: pass
+class ProUser(User): pass
+U = TypeVar('U', bound=User)
+def new_user(user_class: Type[U]) -> U:
+    user = user_class()
```

Comment:
Test invalid args to `__init__`. Test `TypeVar` with non-default `__init__`, and try calling with compatible and incompatible constructor arguments.